app/src/components/ProfileEdit/Question.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react'
+import { useParams, useHistory } from 'react-router'
+import { useHostHomeData } from '../../data/data-context'
+import { ProgressPlugin } from 'webpack'
+import { QuestionType } from '../../models/QuestionType'
+import styled from 'styled-components'
+import {
+    FormControl,
+    FormControlLabel,
+    FormGroup,
+    RadioGroup,
+    Radio,
+    Checkbox,
+    TextField,
+} from '@material-ui/core'
+
+interface Props {
+    question: QuestionType
+    setAnswer: (id: string, value: any) => void
+}
+
+const QuestionLabel = styled.label`
+    margin: 15px 0;
+    display: block;
+`
+
+const Question = (props: Props) => {
+    const { question } = props
+
+    function handleChange(
+        e:
+            | React.ChangeEvent<HTMLInputElement>
+            | React.ChangeEvent<HTMLTextAreaElement>
+    ) {
+        const questionId = props.question.id
+        if (e.target.type === 'checkbox') {
+            if ('checked' in e.target) {
+                props.question.answer[e.target.value] = e.target.checked
+            }
+            props.setAnswer(questionId, props.question.answer)
+        } else {
+            props.setAnswer(questionId, e.target.value)
+        }
+    }
+
+    return (
+        <div>
+            <QuestionLabel>{question.question}</QuestionLabel>
+            <div>
+                {question.type === 'radio' && (
+                    <FormControl component="fieldset">
+                        <RadioGroup
+                            aria-label={question.question}
+                            key={question.id}
+                            name={question.id}
+                            value={props.question.answer}
```

Comment:
good point @tylerthome. instead of putting props.questions,answer directly into the value prop: i would assign the value prop to some state, and have that state value default to null. then update the value prop via state, with the input onchange events driving those updates. hopefully this makes sense, we can discuss further.

i see that answers in the questions dataset have been pre-populated with mock answers - wondering what is reasoning to default to a value other than null?

Reply:
@tylerthome Checkboxes exploded when I set answer to null. I fixed it in a new commit. The rest of the types handle null gracefully.

Reply:
@arshiamasih State is now managed in the QuestionPage component and passed to Question through props.